lib/contracts/daocreator.ts

```diff
@@ -133,32 +133,45 @@ export class DaoCreator extends ExtendTruffleContract(SolidityContract) {
        * scheme will be a contract wrapper
        */
       const scheme = await arcSchemeInfo.contract.at(
-        schemeOptions.address || arcSchemeInfo.address
+        schemeOptions.address || arcSchemeInfo.address,
       );
 
-      let schemeVotingMachineParams = {};
+      let schemeVotingMachineParams = schemeOptions.votingMachineParams;
       let schemeVoteParametersHash;
       let schemeVotingMachine;
 
-      if (schemeOptions.votingMachineParams) {
+      if (schemeVotingMachineParams) {
+        const schemeVotingMachineName = schemeVotingMachineParams.votingMachineName;
+        const schemeVotingMachineAddress = schemeVotingMachineParams.votingMachine;
         /**
-         * get the hash of the voting machine params
+         * get the voting machine contract
          */
-        if (schemeOptions.votingMachineParams.votingMachineName === defaultVotingMachine.votingMachineName) {
+        if (!schemeVotingMachineAddress &&
+          (!schemeVotingMachineName ||
+            (schemeVotingMachineName === defaultVotingMachineParams.votingMachineName))) {
           /**
            *  scheme is using the default voting machine
            */
           schemeVotingMachine = defaultVotingMachine;
         } else {
-          // scheme has its own voting machine. get it.
-          schemeVotingMachine = await Contracts.getScheme(schemeOptions.votingMachineParams.votingMachineName, schemeOptions.votingMachineParams.votingMachine);
-          schemeOptions.votingMachineParams.votingMachine = schemeVotingMachine.address; // in case it wasn't supplied in order to get the default
+          /**
+           * scheme has its own voting machine. Go get it.
```

Comment:
this comment doesnt seem to accurately reflect the conditional below it

Reply:
It follows from the "else" above.  The code following does proceed to get the voting machine.